daemon/main_test.go

```diff
@@ -49,18 +115,58 @@ var _ = g.BeforeSuite(func() {
 })
 
 var _ = g.Describe("Main", func() {
-	g.Context("Connection", func() {
-		dummyPluginDPU := NewDummyPlugin()
-		DpuDaemon := NewDpuDaemon(dummyPluginDPU)
+	var (
+		tmpDir           string
+		err              error
+		serverSocketPath string
+		dpuDaemon        *DpuDaemon
+		hostDaemon       *HostDaemon
+	)
+	g.BeforeEach(func() {
+		tmpDir, err = os.MkdirTemp("", "cniserver")
+		Expect(err).NotTo(HaveOccurred())
+		serverSocketPath = filepath.Join(tmpDir, "server.socket")
 
+		dummyPluginDPU := NewDummyPlugin()
+		dpuDaemon = NewDpuDaemon(dummyPluginDPU)
 		dummyPluginHost := NewDummyPlugin()
-		HostDaemon := NewHostDaemon(dummyPluginHost)
+		m := SriovManagerStub{}
+		hostDaemon = NewHostDaemon(dummyPluginHost).
+			WithCniServerPath(serverSocketPath).
+			WithSriovManager(m)
+	})
+
+	g.AfterEach(func() {
+		klog.Info("Cleaning up")
+		os.RemoveAll(tmpDir)
+		dpuDaemon.Stop()
+		hostDaemon.Stop()
+	})
 
-		g.It("Should connect succesfully if the server is up first", func() {
-			HostDaemon.Start()
-			DpuDaemon.Start()
-			_, err := HostDaemon.CreateBridgePort(1, 1, 1)
-			Expect(err).ShouldNot(HaveOccurred())
+	g.Context("Connection", func() {
+		g.It("Should connect successfully if the DPU Daemon is up first", func() {
```

Comment:
Shouldn't we have another test case to see if Host Daemon is started first?

Reply:
No, not today. In the future, we might want to abstract out using a robust way to communicate. It shouldn't be tested at this level imo.

Reply:
What is the purpose of the test, just to make sure the DPU/Host Daemon can talk to each other?

Reply:
This tests the fact that CNI calls get processed correctly. In the future, we will inject a dependency into HostDaemon to test the cases where we do plumbing.